starter-code/public/javascript/index.js

```diff
@@ -2,15 +2,63 @@ const charactersAPI = new APIHandler('http://localhost:8000');
 
 window.addEventListener('load', () => {
   document.getElementById('fetch-all').addEventListener('click', function (event) {
+    axios.get('http://localhost:8000/characters').then((result) => {
 
+      let resultsFromAPI = result.data
+      console.log(resultsFromAPI)
+
+      resultsFromAPI.forEach((character) => {
+        let newNode = document.getElementsByClassName('character-info')[0].cloneNode(true)
+        newNode.getElementsByClassName('name')[0].innerText = character.name
+        newNode.getElementsByClassName('occupation')[0].innerText = character.occupation
+        newNode.getElementsByClassName('cartoon')[0].innerText = character.cartoon
+        newNode.getElementsByClassName('weapon')[0].innerText = character.weapon
+
+        document.getElementsByClassName('characters-container')[0].append(newNode)
+      })
+
+    })
   });
 
   document.getElementById('fetch-one').addEventListener('click', function (event) {
 
+    let characterId = document.getElementById("character-id").value
+    //need to get the value from the form 
+
+    axios.get(`http://localhost:8000/characters/${characterId}`).then((result) => {
+      let resultsFromAPI = result.data
+      console.log(resultsFromAPI)
+
+      let newNode = document.getElementsByClassName('character-info')[0].cloneNode(true)
+      newNode.getElementsByClassName('name')[0].innerText = resultsFromAPI.name
+      newNode.getElementsByClassName('occupation')[0].innerText = resultsFromAPI.occupation
+      newNode.getElementsByClassName('cartoon')[0].innerText = resultsFromAPI.cartoon
+      newNode.getElementsByClassName('weapon')[0].innerText = resultsFromAPI.weapon
+
+      document.getElementsByClassName('characters-container')[0].append(newNode)
+
+    })
```

Comment:
its also good to do some error handling
this will help you in project module to find your error faster and easie
, so here you should write

.catch(error => console.log("something with grtting charecter id went wrong", error))